### PR TITLE
 [nav] Standardize mRover as a member variable in child state machine classes

### LINKED ISSUE #427: standardize mPhoebe as a member variable in child state machine classes
[nav]

- [ ] standardize use of mPhoebe as a member variable instead of passing a rover variable into functions and child class objects

 ~~decide whether to make isObstacleDetected a StateMachine public member function (visible to child classes) or leave as utilities~~

~~evaluate other functions for possible input changes (e.g. removing the Rover* input pointer)~~

### PR DESCRIPTION
**Pull request by radraju**
Resolve #427

Standardized mRover and mRoverConfig as member variables in searchStateMachine and obstacleAvoidanceStateMachine instead of passing a rover variable into functions and child class objects.

Rename phoebe to rover

This change was tested in the navigation simulator to verify that search and obstacle avoidance worked as usual.

### CODE REVIEW

File: jetson/nav/search/searchStateMachine.cpp [hunk 1]
```diff
@@ -10,46 +10,49 @@
...
 
-// Constructs an SearchStateMachine object with roverStateMachine
-SearchStateMachine::SearchStateMachine(StateMachine* roverStateMachine )
-    : roverStateMachine( roverStateMachine ) {}
+// Constructs an SearchStateMachine object with roverStateMachine, mRoverConfig, and mPhoebe
+SearchStateMachine::SearchStateMachine(StateMachine* roverStateMachine, Rover* rover, const rapidjson::Document& roverConfig)
+    : roverStateMachine( roverStateMachine ) 
+    , mPhoebe( rover ) 
+    , mRoverConfig( roverConfig ) {}
+
```
Comment (ankithu) [inline jetson/nav/search/searchStateMachine.cpp:18]: When I compile I get a warning here. Switching the order of the initializer list so it matches the order of initialization in the hpp file (switch mPhoebe and mRoverConfig) should make the warning go away. I know it doesn't make a difference but its probably just a best practice to not have warnings

File: jetson/nav/obstacle_avoidance/obstacleAvoidanceStateMachine.cpp [hunk 2]
```diff
@@ -6,10 +6,12 @@
...
-// Constructs an ObstacleAvoidanceStateMachine object with roverStateMachine
-ObstacleAvoidanceStateMachine::ObstacleAvoidanceStateMachine( StateMachine* stateMachine_)
-    : roverStateMachine( stateMachine_ ),
-      mJustDetectedObstacle( false ) {}
+// Constructs an ObstacleAvoidanceStateMachine object with roverStateMachine, mRoverConfig, and mPhoebe
+ObstacleAvoidanceStateMachine::ObstacleAvoidanceStateMachine( StateMachine* stateMachine_, Rover* rover, const rapidjson::Document& roverConfig)
+    : roverStateMachine( stateMachine_ )
+    , mPhoebe( rover )
+    , mRoverConfig( roverConfig )
+    ,  mJustDetectedObstacle( false ) {}
```
Comment (ankithu) [inline jetson/nav/obstacle_avoidance/obstacleAvoidanceStateMachine.cpp:14]: When I compile I get a warning here. Switching the order of the initializer list so it matches the order of initialization in the hpp file (switch mPhoebe and mRoverConfig) should make the warning go away. I know it doesn't make a difference but its probably just a best practice to not have warnings

**Review by ankithu — CHANGES_REQUESTED** [review]
See comments, other than those two minor things it seems to work well

File: jetson/nav/obstacle_avoidance/obstacleAvoidanceStateMachine.cpp [hunk 3]
```diff
@@ -6,10 +6,12 @@
 #include <cmath>
 #include <iostream>
 
-// Constructs an ObstacleAvoidanceStateMachine object with roverStateMachine
-ObstacleAvoidanceStateMachine::ObstacleAvoidanceStateMachine( StateMachine* stateMachine_)
-    : roverStateMachine( stateMachine_ ),
-      mJustDetectedObstacle( false ) {}
+// Constructs an ObstacleAvoidanceStateMachine object with roverStateMachine, mRoverConfig, and mPhoebe
+ObstacleAvoidanceStateMachine::ObstacleAvoidanceStateMachine( StateMachine* stateMachine_, Rover* rover, const rapidjson::Document& roverConfig)
```
Comment (aswarner) [inline jetson/nav/obstacle_avoidance/obstacleAvoidanceStateMachine.cpp:10]: add space between parameters and )

File: jetson/nav/obstacle_avoidance/obstacleAvoidanceStateMachine.hpp [hunk 4]
```diff
@@ -12,14 +12,14 @@ enum class ObstacleAvoidanceAlgorithm
...
-// This class is the base class for the logic of the obstacle avoidance state machine
+// This class is the base class for the logic of the obstacle avoidance state machine 
 class ObstacleAvoidanceStateMachine 
 {
 public:
     /*************************************************************************/
     /* Public Member Functions */
     /*************************************************************************/
-    ObstacleAvoidanceStateMachine( StateMachine* stateMachine_ );
+    ObstacleAvoidanceStateMachine( StateMachine* stateMachine_, Rover* rover, const rapidjson::Document& roverConfig);
```
Comment (aswarner) [inline jetson/nav/obstacle_avoidance/obstacleAvoidanceStateMachine.hpp:22]: add space between parameters and )

File: jetson/nav/search/spiralInSearch.hpp [hunk 5]
```diff
@@ -9,8 +9,8 @@
 class SpiralIn : public SearchStateMachine
 {
 public:
-    SpiralIn( StateMachine* stateMachine_ )
-    : SearchStateMachine(stateMachine_) {}
+    SpiralIn( StateMachine* stateMachine_, Rover* rover, const rapidjson::Document& roverConfig )
+    : SearchStateMachine( stateMachine_, rover, roverConfig) {}
```
Comment (aswarner) [inline jetson/nav/search/spiralInSearch.hpp:13]: space after roverConfig

**Review by ankithu — APPROVED** [review]

**Comment by ankithu** [comment]
compiles without warning and tested in sim

**Review by aswarner — DISMISSED** [review]

**Review by DavidSmith166 — DISMISSED** [review]
this is a good start, now you can work on changing the name from phoebe to rover

**Review by ankithu — APPROVED** [review]
Looked over the code, ran it on the sim, looks good to me.

**Review by DavidSmith166 — APPROVED** [review]

**Review by aswarner — APPROVED** [review]

**Pull request merged**